app/models/nearby_facility.rb

```diff
@@ -7,13 +7,47 @@ class << self
     attr_writer :validate_on_load
 
     def query(params)
-      json_from_file = File.read(Rails.root.join('modules', 'va_facilities', 'nearby.json'))
-      mocked_response = JSON.parse(json_from_file)
-      return mocked_response if params[:street_address] && params[:city] && params[:state] && params[:zip]
+      if params[:street_address] && params[:city] && params[:state] && params[:zip]
+        isochrone_response = request_isochrone(params)
+        return get_facilities_in_isochrone(params, isochrone_response)
+      end
 
       NearbyFacility.none
     end
 
+    def request_isochrone(params)
+      params[:drive_time] = '30' unless params[:drive_time]
+      address = "#{params[:street_address]} #{params[:city]} #{params[:state]} #{params[:zip]}"
```

Comment:
Street address feels the most dangerous to me. Right now it looks like we have a placeholder validation of `/\d/`. Do you think there's a way we can go further towards validating an address format that Bing expects? 

A good thing: It looks like no matter how badly you mangle street address, city, and state, it can always return **something** if you have a valid zipcode.

Reply:
Agreed that the `street_address` could be messed up and still easily pass validation at this point. It's the price for not forcing the consumer to geocode the address before making this request. I'd like to look into how the AddressValidator API works to see if it would add more assurance that an address is valid.

I thought this Microsoft response to someone wanting to use Bing Maps API to validate addresses was a nice comment on the predicament:

>Geocoders are not designed to be used as address validators and are not recommended to be used as one . . . Address validators are specialized services which are usually fairly expensive. Many companies use geocoders as a cheap alternative but know that there are many caveats. If you want a true address validator for the US, the US postal service does have an API for this: https://www.usps.com/business/web-tools-apis/welcome.htm

From https://social.msdn.microsoft.com/Forums/lync/en-US/d5c7d15c-3113-482f-a72e-67e5d2aa4dbf/bing-map-address-and-postal-code-validation?forum=bingmaps

I'm adding `investigate stricter address validation` to the list of work still to be done. I'm not sure we need more than we currently have for the MVP, but it would definitely be better to warn the consumer if the "address" they provide is bonkers. This would also allow us to keep bad consumer inputs and actual problems with using the Bing API separate for logging purposes.